packages/server/src/users/api/UserDto.ts

```diff
@@ -0,0 +1,53 @@
+import { ApiProperty, ApiPropertyOptional, OmitType } from '@nestjs/swagger';
+
+import { FavouriteRestaurantDto } from '../../restaurants/api/RestaurantDto';
+import { FavouriteDishDto } from '../../restaurants/dishes/api/DishDto';
+import { ApiObjectIdProperty } from '../../shared';
+
+interface Card {
+  number: number;
```

Comment:
Troszkę wbrew logice tu bym proponował typ string, z kilku powodów:
1. Liczby w JS (i każdym innym języku, pomijając konstrukcje typu BigInt/Decimal) mają ograniczoną precyzję i przy tak dużych liczbach jak numery kart zaczynają świrować, wpisz w konsolę to i zobacz sobie wynik: `9007_1992_5474_0992 === 9007_1992_5474_0993` (to tylko liczba 16-cyfrowa, a według wikipedii numery kart mogą dochodzić aż do 19).
2. Trudno ogarnia się zera wiodące w typach numerycznych, np: chowając w bazie liczbę 1 nie sposób potem rozszyfrować, czy to oryginalnie było `1`, `01`, `001`, `0001` itd.
3. Nie będziemy wykonywać na nich żadnych operacji numerycznych - nie wyobrażam sobie sytuacji w której robimy `number + 1` czy coś takiego.

Z podobnych powodów robi się tak też z PESELami itd.

Reply:
super wytłumaczone, dzięki ! :)